### PR TITLE
[PERFORMANCE] A collection of minor performance enhancements

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/505/commits/2356b2164c938ef0600f804d26d37c62de6015da

## Before

`Message$Builder.build` takes 11.4% of MIME4J parsing time, 10% of MIME4J memory allocation (for the header parsing use case):

Screenshot from 2021-06-22 07-45-42

## After

Screenshot from 2021-06-22 07-48-27

Disappeared!

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/505/commits/61c63418dfcff2f6003dda0b7ac26d096c2bd684

## Before

Screenshot from 2021-06-22 07-50-14

We can clearly see getUserFlags is called twice.

## After

Not anmore.

Screenshot from 2021-06-22 07-51-48

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/505/commits/9d113fb78f543e8f7beb312a8f5462093407ad80

## Before

Screenshot from 2021-06-22 07-53-53

Takes 0.2 % of CPU and 0.5% of memory allocation.

## After

Screenshot from 2021-06-22 07-56-01

Slimmer it now takes just 0.14% of CPU and only 0.25% of memory... Not game changer but nice to take.

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/505/commits/0cd70cfde85f12f8cd9b7bbef54cc7215b9337ba

## Before

0.2% of CPU, 0.34% of memory allocation.

Screenshot from 2021-06-22 07-58-20

## After

No traces :-)

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/505/commits/5c7ab34fe3eda088e4a99930b2c79aa754d60084

## Before

Screenshot from 2021-06-22 08-01-45

0.12% of CPU, 0.09% of memory

## After

No traces :-)

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageViewFactory.java [hunk 1]
```diff
@@ -80,6 +80,9 @@ static void assertOneMessageId(Collection<MessageResult> messageResults) {
         }
 
         private static boolean hasOnlyOneMessageId(Collection<MessageResult> messageResults) {
+            if (messageResults.size() <= 1) {
```
Comment (rouazana) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageViewFactory.java:83]: you mean `if (messageResults.size() == 1)`

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Comment by chibenwa** [comment]
Just squashed.

**Pull request merged**